controllers/componentdetectionquery_controller.go

```diff
@@ -126,6 +129,15 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 		if context == "" {
 			context = "./"
 		}
+		if source.Revision == "" {
+			log.Info(fmt.Sprintf("Look for default branch of repo %s... %v", source.URL, req.NamespacedName))
+			source.Revision, err = github.GetDefaultBranchFromRepo(source.URL, r.GitHubClient, ctx)
+			if err != nil {
```

Comment:
Rather than erroring out here, wonder if we should just the log the error and default to `main`?

Reply:
updated to fall back to use `main`, and check for existence before setting it. 
as discussed, if error occurred when try to look for `defaultBranch`, and `main` branch does not exist. we error out